errors/errors.go

```diff
@@ -0,0 +1,110 @@
+package errors
```

Comment:
This package needs tests, at least for New and Is, and maybe one of the convenience methods.